### ISSUE TITLE
When converting from Typst, `primes` are handled incorrectly

### ISSUE THREAD

**Issue by John-Toohey** [jgm/pandoc#10411]
Consider the following Typst file:

```typst
$ f'(x) = x^2 $
```
Which renders as: 
![image](https://github.com/user-attachments/assets/c7154418-cb2f-491e-8db6-c05544a3968d)

When run through pandoc `pandoc a.typ -o a.md` the following output is generated:

```markdown
$$f\prime(x) = x^{2}$$
```

Which, when converted to PDF via pdflatex, looks like this --- with an incredibly chunky prime sign:

![image](https://github.com/user-attachments/assets/0fdf03ba-05f9-4b61-adbc-60b83efbcdb4)

This is incorrect. Rather the markdown should be (notice the superscript):

```md
$$f^{\prime}(x) = x^{2}$$
```

**Pandoc version?**
pandoc 3.1.11.1 running on NixOS unstable. Also tested on https://pandoc.org/try/

**Comment by alerque** [comment]
This is not correct, you should not need to manually superscript the `\prime` in LaTeX. This is a function of the font not fully supporting the expected layout features.

Another iteration of this problem can be seen with layout engines (including Typst) that try to over-compensate and move something that the font has already made into a superscript manually smaller and in a higher position resulting in a doubly small and high superscirpt. I believe this has been at least partially addressed in the latest Typst devel version, but the last stable release still shows this problem, e.g. with *Libertinus Math*. You can see the effect here:

https://polytype.dev/prime-symbol/

**Comment by jgm** [comment]
Really we should be going for:
```
$$f'(x) = x^{2}$$
```
which is rendered correctly.

**Comment by jgm** [comment]
This needs to be addressed in `texmath`; moving there.
